.circleci/config.yml

```diff
@@ -0,0 +1,278 @@
+      python-version:
+        type: string
+    steps:
+      - run:
+          name: Setup Win environment
+          command: |
+            conda init powershell
+            conda create -y -n py-<< parameters.python-version >> python=<< parameters.python-version >>
+            conda activate py-<< parameters.python-version >>
+            python -m venv py_venv
+
+  bootstrap-linux:
+    steps:
+      - run:
+          name: Setup Linux environment
+          command: |
+            sudo apt-get update
+            sudo apt-get install \
+            -y cmake g++ python3 python3-pip git 
+            python -m venv py_venv
+
+  install-python-deps:
+    steps:
+      - run:
+          name: Install Python dependencies
+          command: |
+            . py_venv/bin/activate
+            python --version
+            python -m pip install --upgrade pip
+            python -m pip install -r requirements.txt
+            python -m pip install --force-reinstall -r requirements-dev.txt
+            python -m pip install cmake flake8 bandit setuptools-scm 
+            python -m pip install coverage codacy-coverage
+
+  python-build-test:
+    steps:
+      - run:
+          name: Build & test python
+          command: |
+            . py_venv/bin/activate
+            python --version
+            python -m pip freeze
+            python setup.py build_ext --inplace
+            python setup.py test --verbose
+
+  flake8-req-check:
+    steps:
+      - run:
+          name: Check code errors (pyflakes)
+          command: |
+            . py_venv/bin/activate
+            flake8 --ignore=E --per-file-ignores="__init__.py:F401" --exclude=py_venv .
+
+  flake8-opt-check:
+    steps:
+      - run:
+          name: Check formatting rules (PEP8)
+          command: |
+            . py_venv/bin/activate
+            git diff HEAD^ | flake8 --diff --per-file-ignores="__init__.py:F401"
+
+  bandit-check:
+    steps:
+      - run:
+          name: Bandit security check
+          command: |
+            . py_venv/bin/activate
+            bandit -c bandit.yml -r camille
+
+  codacy-check:
+    steps:
+      - run:
+          name: Codacy code coverage check
+          command: |
+            . py_venv/bin/activate
+            if [ ! -z "${CODACY_PROJECT_TOKEN}" ]; then 
+              coverage run setup.py test;
+              coverage xml;
+              python-codacy-coverage -r coverage.xml;
+            fi
+
+  create-sdist:
+    steps:
+      - run:
+          name: Create source package
+          command: |
+            . py_venv/bin/activate
+            python -m pip install twine
+            python setup.py sdist
+
+  upload-pypi:
+    steps:
+      - run:
+          name: Upload to pypi
+          command: |
+            . py_venv/bin/activate
+            twine upload  --repository testpypi --skip-existing dist/*
+
+jobs:
+  python-check:
+    parameters:
+      version:
+        type: string
+        default: "3.6"
+      formatting:
```

Comment:
Defining default makes the parameter optional. Otherwise the parameter is taken as required. In that case, missing parameter is identified as config file error and CircleCI will stop running.